packages/router/src/Exceptions/HttpExceptionHandler.php

```diff
@@ -2,73 +2,54 @@
-use Tempest\Http\Session\CsrfTokenDidNotMatch;
 use Tempest\Http\Status;
 use Tempest\Router\ResponseSender;
-use Tempest\Support\Filesystem;
-use Tempest\View\GenericView;
+use Tempest\Router\RouteConfig;
+use Tempest\Support\Arr;
 use Throwable;
 
 final readonly class HttpExceptionHandler implements ExceptionHandler
 {
     public function __construct(
-        private AppConfig $appConfig,
-        private Kernel $kernel,
         private ResponseSender $responseSender,
+        private Kernel $kernel,
         private Container $container,
-        private ExceptionReporter $exceptionReporter,
+        private ExceptionProcessor $exceptionProcessor,
+        private RouteConfig $routeConfig,
     ) {}
 
     public function handle(Throwable $throwable): void
     {
-        try {
-            $this->exceptionReporter->report($throwable);
+        $request = $this->container->get(Request::class);
 
-            $response = match (true) {
-                $throwable instanceof ConvertsToResponse => $throwable->toResponse(),
-                $throwable instanceof AccessWasDenied => $this->renderErrorResponse(Status::FORBIDDEN),
-                $throwable instanceof HttpRequestFailed => $this->renderErrorResponse($throwable->status, $throwable),
-                $throwable instanceof CsrfTokenDidNotMatch => $this->renderErrorResponse(Status::UNPROCESSABLE_CONTENT),
-                default => $this->renderErrorResponse(Status::INTERNAL_SERVER_ERROR),
-            };
-
-            $this->responseSender->send($response);
+        try {
+            $this->exceptionProcessor->process($throwable);
+            $this->responseSender->send($this->renderResponse($request, $throwable));
         } finally {
             $this->kernel->shutdown();
         }
     }
 
-    private function renderErrorResponse(Status $status, ?HttpRequestFailed $exception = null): Response
+    public function renderResponse(Request $request, Throwable $throwable): Response
     {
-        return new GenericResponse(
-            status: $status,
-            body: new GenericView(__DIR__ . '/HttpErrorResponse/error.view.php', [
-                'css' => $this->getStyleSheet(),
-                'status' => $status->value,
-                'title' => $status->description(),
-                'message' => $exception?->getMessage() ?: match ($status) {
-                    Status::INTERNAL_SERVER_ERROR => 'An unexpected server error occurred',
-                    Status::NOT_FOUND => 'This page could not be found on the server',
-                    Status::FORBIDDEN => 'You do not have permission to access this page',
-                    Status::UNAUTHORIZED => 'You must be authenticated in to access this page',
-                    Status::UNPROCESSABLE_CONTENT => 'The request could not be processed due to invalid data',
-                    default => null,
-                },
-            ]),
-        );
-    }
+        ksort($this->routeConfig->exceptionRenderers);
```

Comment:
Shouldn't we sort during discovery instead?

Reply:
I'd rather sort as the last step in case a renderer is added during runtime for any reason